app/cli/commandlineparser.cpp

```diff
@@ -0,0 +1,335 @@
+#include "commandlineparser.h"
+
+#include <QCommandLineParser>
+#include <QRegularExpression>
+
+#if defined(Q_OS_WIN)
+#include <qt_windows.h>
+#endif
+
+static bool inRange(int value, int min, int max)
+{
+    return value >= min && value <= max;
+}
+
+class CommandLineParser : public QCommandLineParser
```

Comment:
Can you making the argument parsing in this class case-insensitive?

Reply:
Positional arguments should be trivial to handle in case insensitive manner. Same for option values.

However, QCommandLineParser doesn't seem to provide any easy to way to parse option names in case insensitive manner. But perhaps unknownOptionNames() could be used to detect any options that were given in wrong case. And then adjust them to correct case and reparse them... Will have to try.

Reply:
Ok